docs/kubeconfig-file.md

```diff
@@ -84,15 +84,17 @@ If the contents of the kubernetes_auth file conflict with entries in .kubeconfig
 
 ## Loading and merging rules
 The rules for loading and merging the .kubeconfig files are straightforward, but there are a lot of them.  The final config is built in this order:
-  1.  Merge together the kubeconfig itself.  This is done with the following hierarchy and merge rules:
+  1.  Get the kubeconfig  from disk.  This is done with the following hierarchy and merge rules:
+      
 
+      If the CommandLineLocation (the value of the `kubeconfig` command line option) is set, use this file only.  No merging.  Otherwise:
+
+      Merge files together based on the following rules.
       Empty filenames are ignored.  Files with non-deserializable content produced errors.
       The first file to set a particular value or map key wins and the value or map key is never changed.
       This means that the first file to set CurrentContext will have its context preserved.  It also means that if two files specify a "red-user", only values from the first file's red-user are used.  Even non-conflicting entries from the second file's "red-user" are discarded.
-      1.  CommandLineLocation - the value of the `kubeconfig` command line option
-      1.  EnvVarLocation - the value of $KUBECONFIG
       1.  CurrentDirectoryLocation - ``pwd``/.kubeconfig
-      1.  HomeDirectoryLocation = ~/.kube/.kubeconfig
+      1.  If EnvVarLocation (the value of $KUBECONFIG) is available, use it.  Otherwise, HomeDirectoryLocation (~/.kube/config).
```

Comment:
Do you think it's going to be common to want to provide a list of files?  I don't imagine I would suggest that anyone other than a power user (due to aforementioned complexity with merging explanation).  Would there be a scenario where list-of-files via --config would be needed if you could always specify list-of-files via ENV?

----- Original Message -----
> >        1.  CurrentDirectoryLocation - ``pwd``/.kubeconfig
> > -      1.  HomeDirectoryLocation = ~/.kube/.kubeconfig
> > +      1.  If EnvVarLocation (the value of $KUBECONFIG) is available, use
> > it.  Otherwise, HomeDirectoryLocation (~/.kube/config).
> 
> That could be accomplished by the list-of-files approach if relative paths
> were allowed.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/6680/files#r28347219